### ISSUE TITLE
tendermint v0.35 spamming log with p2p connection ERRORs

### ISSUE THREAD

**Issue by p4u** [vocdoni/vocdoni-node#641]
Since we migrated to Tendermint 0.35 this message appears in most of the nodes.

The current branch is dev which is in sync with bizono-0.1

```
2022-10-26T10:21:09Z ERROR failed to handshake with peer err="expected to connect with peer \"421af4784cceea91db0718c30b87d4dce4b2a7c2\", got \"60175ea102f98131246caf1851c650874e4b5231\"" module=p2p peer={"Hostname":"95.217.14.22","NodeID":"421af4784cceea91db0718c30b87d4dce4b2a7c2","Path":"","Port":26656,"Protocol":"mconn"}
2022-10-26T10:21:11Z ERROR failed to handshake with peer err="expected to connect with peer \"6b7e3696ac946e85e897da256866a63436528980\", got \"99c97c6d634da42d7cac9f9e66a644f169743e14\"" module=p2p peer={"Hostname":"78.46.243.40","NodeID":"6b7e3696ac946e85e897da256866a63436528980","Path":"","Port":26656,"Protocol":"mconn"}
2022-10-26T10:21:11Z ERROR failed to handshake with peer err="expected to connect with peer \"4432e41ec289d4ad3105a4a62a4568fbdcbc2f39\", got \"574bc64f9643d96e501c0d7fc71c802202b29dad\"" module=p2p peer={"Hostname":"135.181.92.20","NodeID":"4432e41ec289d4ad3105a4a62a4568fbdcbc2f39","Path":"","Port":26656,"Protocol":"mconn"}
2022-10-26T10:21:12Z ERROR failed to handshake with peer err="expected to connect with peer \"0c46950d1f35829cf65fddba341ea8a984de2429\", got \"fae35a69f7b684fabf8d23d3d6d734341611c5f6\"" module=p2p peer={"Hostname":"95.217.8.76","NodeID":"0c46950d1f35829cf65fddba341ea8a984de2429","Path":"","Port":26656,"Protocol":"mconn"}
2022-10-26T10:21:12Z ERROR failed to handshake with peer err="expected to connect with peer \"421af4784cceea91db0718c30b87d4dce4b2a7c2\", got \"60175ea102f98131246caf1851c650874e4b5231\"" module=p2p peer={"Hostname":"95.217.14.22","NodeID":"421af4784cceea91db0718c30b87d4dce4b2a7c2","Path":"","Port":26656,"Protocol":"mconn"}
2022-10-26T10:21:14Z ERROR failed to handshake with peer err="expected to connect with peer \"280d70e76587195fcbd82ebb732b1b7987676058\", got \"afc86a79e7c7b498e488be94245a4ee02e3416f8\"" module=p2p peer={"Hostname":"135.181.155.219","NodeID":"280d70e76587195fcbd82ebb732b1b7987676058","Path":"","Port":26656,"Protocol":"mconn"}
```

**Comment by altergui** [comment]
i've been looking into this yesterday. here's the list of peers for `DVOTE_VOCHAIN=dev`

```
{5e083352c539d43f9713236e8ff9f0045dca0822 mconn://5e083352c539d43f9713236e8ff9f0045dca0822@116.203.57.31:26656}
{135cb4488595d2d2ae15a687ba0ce45036ee41bc mconn://135cb4488595d2d2ae15a687ba0ce45036ee41bc@116.203.88.124:26656}
{280d70e76587195fcbd82ebb732b1b7987676058 mconn://280d70e76587195fcbd82ebb732b1b7987676058@135.181.155.219:26656}
{afc86a79e7c7b498e488be94245a4ee02e3416f8 mconn://afc86a79e7c7b498e488be94245a4ee02e3416f8@135.181.155.219:26656}
{4432e41ec289d4ad3105a4a62a4568fbdcbc2f39 mconn://4432e41ec289d4ad3105a4a62a4568fbdcbc2f39@135.181.92.20:26656}
{574bc64f9643d96e501c0d7fc71c802202b29dad mconn://574bc64f9643d96e501c0d7fc71c802202b29dad@135.181.92.20:26656}
{3bfa4a2b752bace77664579d75a64c502ed21596 mconn://3bfa4a2b752bace77664579d75a64c502ed21596@135.181.92.22:26656}
{69e9e1ba895304ddddd6cc2ea7f6c3b16a223863 mconn://69e9e1ba895304ddddd6cc2ea7f6c3b16a223863@135.181.92.30:26656}
{b346c87fb39d4f1999405957c2ff7edc7b95e80d mconn://b346c87fb39d4f1999405957c2ff7edc7b95e80d@188.34.177.219:26656}
{7440a5b086e16620ce7b13198479016aa2b07988 mconn://7440a5b086e16620ce7b13198479016aa2b07988@65.108.62.6:26656}
{6b7e3696ac946e85e897da256866a63436528980 mconn://6b7e3696ac946e85e897da256866a63436528980@78.46.243.40:26656}
{99c97c6d634da42d7cac9f9e66a644f169743e14 mconn://99c97c6d634da42d7cac9f9e66a644f169743e14@78.46.243.40:26656}
{a366c3a770da23e43012722a354a74aca8031f6f mconn://a366c3a770da23e43012722a354a74aca8031f6f@91.126.18.128:26656}
{421af4784cceea91db0718c30b87d4dce4b2a7c2 mconn://421af4784cceea91db0718c30b87d4dce4b2a7c2@95.217.14.22:26656}
{60175ea102f98131246caf1851c650874e4b5231 mconn://60175ea102f98131246caf1851c650874e4b5231@95.217.14.22:26656}
{0c46950d1f35829cf65fddba341ea8a984de2429 mconn://0c46950d1f35829cf65fddba341ea8a984de2429@95.217.8.76:26656}
{fae35a69f7b684fabf8d23d3d6d734341611c5f6 mconn://fae35a69f7b684fabf8d23d3d6d734341611c5f6@95.217.8.76:26656}
```

the ips that are publishing two different nodes (on the same port) are the ones generating the error. i don't know how the `dev` deployment is architected but my hunch is that there's a forgotten docker process in each of those nodes.

**Comment by altergui** [comment]
so, i discovered one of those "phantom" nodes was due to this env var in `seed.dev.vocdoni.net`
```
vocdoni@seed:~/dvotenode$ grep SEED env
#DVOTE_VOCHAINCONFIG_SEEDS=
DVOTE_VOCHAINCONFIG_SEEDS=4432e41ec289d4ad3105a4a62a4568fbdcbc2f39@miner1.dev.vocdoni.net:26656
```

i removed it and restarted that node docker

the other 4 phantom nodes are gw{1,2,3}.dev.vocdoni.net and oracle.dev.vocdoni.net
i haven't found yet where they are coming from. maybe they are cached somehow by this same seed node? but i doubt it.

there are 14 peers, more than i see in monitor.vocdoni.net, so i guess those other "rogue" nodes are the ones propagating those "phantom" node ids, probably due to something hardcoded in those rogue nodes env.

**Comment by altergui** [comment]
while chasing this, came across a small bug and pushed a hotfix #660

**Comment by altergui** [comment]
i'm closing this as WONTFIX
these errors remain, and for example one of them comes from 91.126.18.128 which resolves to cli-5b7e1280.wholesale.adamo.es, where i have no access to.

in fact, anyone can put up a rogue node online, publish a phantom peer and this will cause more errors in the log, which we can't control at all

```
 ERROR failed to handshake with peer err="expected to connect with peer \"0c46950d1f35829cf65fddba341ea8a984de2429\", got \"fae35a69f7b684fabf8d23d3d6d734341611c5f6\"" module=p2p peer={"Hostname":"95.217.8.76","NodeID":"0c46950d1f35829cf65fddba341ea8a984de2429","Path":"","Port":26656,"Protocol":"mconn"}
 ERROR failed to handshake with peer err="expected to connect with peer \"280d70e76587195fcbd82ebb732b1b7987676058\", got \"afc86a79e7c7b498e488be94245a4ee02e3416f8\"" module=p2p peer={"Hostname":"135.181.155.219","NodeID":"280d70e76587195fcbd82ebb732b1b7987676058","Path":"","Port":26656,"Protocol":"mconn"}
 ERROR failed to handshake with peer err="expected to connect with peer \"2b2fb3116865ee3792df759a9bd702a376749a04\", got \"a366c3a770da23e43012722a354a74aca8031f6f\"" module=p2p peer={"Hostname":"91.126.18.128","NodeID":"2b2fb3116865ee3792df759a9bd702a376749a04","Path":"","Port":26656,"Protocol":"mconn"}
 ERROR failed to handshake with peer err="expected to connect with peer \"421af4784cceea91db0718c30b87d4dce4b2a7c2\", got \"60175ea102f98131246caf1851c650874e4b5231\"" module=p2p peer={"Hostname":"95.217.14.22","NodeID":"421af4784cceea91db0718c30b87d4dce4b2a7c2","Path":"","Port":26656,"Protocol":"mconn"}
 ERROR failed to handshake with peer err="expected to connect with peer \"4432e41ec289d4ad3105a4a62a4568fbdcbc2f39\", got \"574bc64f9643d96e501c0d7fc71c802202b29dad\"" module=p2p peer={"Hostname":"135.181.92.20","NodeID":"4432e41ec289d4ad3105a4a62a4568fbdcbc2f39","Path":"","Port":26656,"Protocol":"mconn"}
 ERROR failed to handshake with peer err="expected to connect with peer \"6b7e3696ac946e85e897da256866a63436528980\", got \"99c97c6d634da42d7cac9f9e66a644f169743e14\"" module=p2p peer={"Hostname":"78.46.243.40","NodeID":"6b7e3696ac946e85e897da256866a63436528980","Path":"","Port":26656,"Protocol":"mconn"}
```

**Comment by altergui** [comment]
i dug further into this, and it turns out tendermint v0.35 introduced all these logger.Error() calls which were simply not there in v0.34
all of this handshake failures happened as well, they were simply not logged as errors.

https://github.com/tendermint/tendermint/issues/6987

unfortunately there's currently no way in tendermint log package to customize, for example, that p2p module doesn't log any output, but other modules do.
